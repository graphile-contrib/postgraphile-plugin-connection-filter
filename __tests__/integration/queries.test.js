const { graphql } = require("graphql");
const { withPgClient } = require("../helpers");
const { createPostGraphileSchema } = require("postgraphile-core");
const { readdirSync, readFile: rawReadFile } = require("fs");
const { resolve: resolvePath } = require("path");
const { printSchema } = require("graphql/utilities");
const debug = require("debug")("graphile-build:schema");

function readFile(filename, encoding) {
  return new Promise((resolve, reject) => {
    rawReadFile(filename, encoding, (err, res) => {
      if (err) reject(err);
      else resolve(res);
    });
  });
}

const queriesDir = `${__dirname}/../fixtures/queries`;
const queryFileNames = readdirSync(queriesDir);
let queryResults = [];

const kitchenSinkData = () => readFile(`${__dirname}/../p-data.sql`, "utf8");

beforeAll(() => {
  // Get a few GraphQL schema instance that we can query.
  const gqlSchemasPromise = withPgClient(async pgClient => {
    // Different fixtures need different schemas with different configurations.
    // Make all of the different schemas with different configurations that we
    // need and wait for them to be created in parallel.
    const [
      normal,
      dynamicJson,
      relations,
      simpleCollections,
      nullAndEmptyAllowed,
    ] = await Promise.all([
      createPostGraphileSchema(pgClient, ["p"], {
        skipPlugins: [require("graphile-build-pg").PgConnectionArgCondition],
        appendPlugins: [require("../../index.js")],
      }),
      createPostGraphileSchema(pgClient, ["p"], {
        skipPlugins: [require("graphile-build-pg").PgConnectionArgCondition],
        appendPlugins: [require("../../index.js")],
        dynamicJson: true,
      }),
      createPostGraphileSchema(pgClient, ["p"], {
        skipPlugins: [require("graphile-build-pg").PgConnectionArgCondition],
        appendPlugins: [require("../../index.js")],
        graphileBuildOptions: {
          connectionFilterRelations: true,
        },
      }),
      createPostGraphileSchema(pgClient, ["p"], {
        skipPlugins: [require("graphile-build-pg").PgConnectionArgCondition],
        appendPlugins: [require("../../index.js")],
        simpleCollections: "only",
      }),
      createPostGraphileSchema(pgClient, ["p"], {
        skipPlugins: [require("graphile-build-pg").PgConnectionArgCondition],
        appendPlugins: [require("../../index.js")],
        graphileBuildOptions: {
          connectionFilterAllowNullInput: true,
          connectionFilterAllowEmptyObjectInput: true,
        },
      }),
    ]);
    debug(printSchema(normal));
    return {
      normal,
      dynamicJson,
      relations,
      simpleCollections,
      nullAndEmptyAllowed,
    };
  });

  // Execute all of the queries in parallel. We will not wait for them to
  // resolve or reject. The tests will do that.
  //
  // All of our queries share a single client instance.
  const queryResultsPromise = (async () => {
    // Wait for the schema to resolve. We need the schema to be introspected
    // before we can do anything else!
    const gqlSchemas = await gqlSchemasPromise;
    // Get a new Postgres client instance.
    return await withPgClient(async pgClient => {
      // Add data to the client instance we are using.
      await pgClient.query(await kitchenSinkData());
      // Run all of our queries in parallel.
      return await Promise.all(
        queryFileNames.map(async fileName => {
          // Read the query from the file system.
          const query = await readFile(
            resolvePath(queriesDir, fileName),
            "utf8"
          );
          // Get the appropriate GraphQL schema for this fixture. We want to test
          // some specific fixtures against a schema configured slightly
          // differently.
          const schemas = {
            "connections-filter.dynamic-json.graphql": gqlSchemas.dynamicJson,
            "connections-filter.relations.graphql": gqlSchemas.relations,
            "connections-filter.simple-collections.graphql":
              gqlSchemas.simpleCollections,
            "connections-filter.null-and-empty-allowed.graphql":
              gqlSchemas.nullAndEmptyAllowed,
          };
          const gqlSchema = schemas[fileName]
            ? schemas[fileName]
            : gqlSchemas.normal;
          // Return the result of our GraphQL query.
          const result = await graphql(gqlSchema, query, null, {
            pgClient: pgClient,
          });
          if (result.errors) {
            console.log(result.errors.map(e => e.originalError));
          }
          return result;
        })
      );
    });
  })();

  // Flatten out the query results promise.
  queryResults = queryFileNames.map(async (_, i) => {
    return await (await queryResultsPromise)[i];
  });
});

for (let i = 0; i < queryFileNames.length; i++) {
  test(queryFileNames[i], async () => {
    expect(await queryResults[i]).toMatchSnapshot();
  });
}
