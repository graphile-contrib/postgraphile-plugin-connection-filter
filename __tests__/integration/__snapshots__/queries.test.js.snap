// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connections-filter.dynamic-json.graphql 1`] = `
Object {
  "data": Object {
    "a": Object {
      "nodes": Array [
        Object {
          "id": 4,
          "jsonb": Object {
            "boolean": false,
            "int": 4,
            "string": "test",
          },
        },
      ],
    },
    "b": Object {
      "nodes": Array [
        Object {
          "id": 4,
          "jsonb": Object {
            "boolean": false,
            "int": 4,
            "string": "test",
          },
        },
      ],
    },
    "c": Object {
      "nodes": Array [
        Object {
          "id": 4,
          "jsonb": Object {
            "boolean": false,
            "int": 4,
            "string": "test",
          },
        },
      ],
    },
  },
}
`;

exports[`connections-filter.graphql 1`] = `
Object {
  "data": Object {
    "allFilterables_computedSetofChild_name_equalTo_child2": Object {
      "nodes": Array [
        Object {
          "computedSetofChild": Object {
            "nodes": Array [
              Object {
                "id": 2,
                "name": "child2",
              },
            ],
          },
          "id": 1,
        },
        Object {
          "computedSetofChild": Object {
            "nodes": Array [],
          },
          "id": 2,
        },
        Object {
          "computedSetofChild": Object {
            "nodes": Array [],
          },
          "id": 3,
        },
        Object {
          "computedSetofChild": Object {
            "nodes": Array [],
          },
          "id": 4,
        },
        Object {
          "computedSetofChild": Object {
            "nodes": Array [],
          },
          "id": 5,
        },
      ],
    },
    "boolean_distinctFrom_true": Object {
      "nodes": Array [
        Object {
          "boolean": false,
          "id": 3,
        },
        Object {
          "boolean": false,
          "id": 4,
        },
        Object {
          "boolean": null,
          "id": 5,
        },
      ],
    },
    "boolean_equalTo_true": Object {
      "nodes": Array [
        Object {
          "boolean": true,
          "id": 1,
        },
        Object {
          "boolean": true,
          "id": 2,
        },
      ],
    },
    "boolean_isNull_false": Object {
      "nodes": Array [
        Object {
          "boolean": true,
          "id": 1,
        },
        Object {
          "boolean": true,
          "id": 2,
        },
        Object {
          "boolean": false,
          "id": 3,
        },
        Object {
          "boolean": false,
          "id": 4,
        },
      ],
    },
    "boolean_isNull_true": Object {
      "nodes": Array [
        Object {
          "boolean": null,
          "id": 5,
        },
      ],
    },
    "boolean_notDistinctFrom_true": Object {
      "nodes": Array [
        Object {
          "boolean": true,
          "id": 1,
        },
        Object {
          "boolean": true,
          "id": 2,
        },
      ],
    },
    "boolean_notEqualTo_true": Object {
      "nodes": Array [
        Object {
          "boolean": false,
          "id": 3,
        },
        Object {
          "boolean": false,
          "id": 4,
        },
      ],
    },
    "computedIntArray_equalTo_2_20": Object {
      "nodes": Array [
        Object {
          "boolean": true,
          "id": 2,
        },
      ],
    },
    "computedIntArray_equalTo_empty": Object {
      "nodes": Array [],
    },
    "computed_equalTo_test_computed": Object {
      "nodes": Array [
        Object {
          "boolean": false,
          "id": 4,
        },
      ],
    },
    "enum_equalTo_OK": Object {
      "nodes": Array [
        Object {
          "enum": "OK",
          "id": 2,
        },
      ],
    },
    "filterableById_1_computedSetofChild_name_equalTo_child2": Object {
      "computedSetofChild": Object {
        "nodes": Array [
          Object {
            "id": 2,
            "name": "child2",
          },
        ],
      },
      "id": 1,
    },
    "inet_containedBy": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "inet": "192.168.0.1",
        },
      ],
    },
    "inet_containedByOrEqualTo_ContainedBy": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "inet": "192.168.1.1",
        },
      ],
    },
    "inet_containedByOrEqualTo_EqualTo": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "inet": "192.168.1.1",
        },
      ],
    },
    "inet_contains": Object {
      "nodes": Array [
        Object {
          "id": 3,
          "inet": "10.0.0.0/24",
        },
      ],
    },
    "inet_containsOrContainedBy_ContainedBy": Object {
      "nodes": Array [
        Object {
          "id": 3,
          "inet": "10.0.0.0/24",
        },
      ],
    },
    "inet_containsOrContainedBy_Contains": Object {
      "nodes": Array [
        Object {
          "id": 4,
          "inet": "172.168.1.1",
        },
      ],
    },
    "inet_containsOrEqualTo_Contains": Object {
      "nodes": Array [
        Object {
          "id": 3,
          "inet": "10.0.0.0/24",
        },
      ],
    },
    "inet_containsOrEqualTo_EqualTo": Object {
      "nodes": Array [
        Object {
          "id": 3,
          "inet": "10.0.0.0/24",
        },
      ],
    },
    "intArray_anyEqualTo_2": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
      ],
    },
    "intArray_anyGreaterThanOrEqualTo_2": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_anyGreaterThan_2": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_anyLessThanOrEqualTo_2": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
      ],
    },
    "intArray_anyLessThan_2": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
      ],
    },
    "intArray_anyNotEqualTo_2": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_containedBy_2": Object {
      "nodes": Array [],
    },
    "intArray_containedBy_2_3_20": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
      ],
    },
    "intArray_containedBy_empty": Object {
      "nodes": Array [],
    },
    "intArray_contains_2": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
      ],
    },
    "intArray_contains_2_3_20": Object {
      "nodes": Array [],
    },
    "intArray_contains_empty": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_distinctFrom_2_20": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
        Object {
          "id": 5,
          "intArray": null,
        },
      ],
    },
    "intArray_distinctFrom_empty": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
        Object {
          "id": 5,
          "intArray": null,
        },
      ],
    },
    "intArray_equalTo_2_20": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
      ],
    },
    "intArray_equalTo_empty": Object {
      "nodes": Array [],
    },
    "intArray_greaterThanOrEqualTo_2_20": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_greaterThanOrEqualTo_empty": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_greaterThan_2_20": Object {
      "nodes": Array [
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_greaterThan_empty": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_isNull_false": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_isNull_true": Object {
      "nodes": Array [
        Object {
          "id": 5,
          "intArray": null,
        },
      ],
    },
    "intArray_lessThanOrEqualTo_2_20": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
      ],
    },
    "intArray_lessThanOrEqualTo_empty": Object {
      "nodes": Array [],
    },
    "intArray_lessThan_2_20": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
      ],
    },
    "intArray_lessThan_empty": Object {
      "nodes": Array [],
    },
    "intArray_notDistinctFrom_2_20": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
      ],
    },
    "intArray_notDistinctFrom_empty": Object {
      "nodes": Array [],
    },
    "intArray_notEqualTo_2_20": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_notEqualTo_empty": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "intArray": Array [
            1,
            10,
          ],
        },
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
        Object {
          "id": 4,
          "intArray": Array [
            4,
            40,
          ],
        },
      ],
    },
    "intArray_overlaps_2": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
      ],
    },
    "intArray_overlaps_2_3_20": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "intArray": Array [
            2,
            20,
          ],
        },
        Object {
          "id": 3,
          "intArray": Array [
            3,
            30,
          ],
        },
      ],
    },
    "intArray_overlaps_empty": Object {
      "nodes": Array [],
    },
    "int_equalTo_2_and_string_endsWith_t": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
      ],
    },
    "int_equalTo_2_or_string_endsWith_t": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "int_greaterThanOrEqualTo_2": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "int_greaterThan_2": Object {
      "nodes": Array [
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "int_in_2_3": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
      ],
    },
    "int_in_empty": Object {
      "nodes": Array [],
    },
    "int_lessThanOrEqualTo_2": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
      ],
    },
    "int_lessThan_2": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
      ],
    },
    "int_notIn_2_3": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "int_notIn_empty": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
        Object {
          "id": 5,
          "string": null,
        },
      ],
    },
    "jsonb_containedBy_object": Object {
      "nodes": Array [
        Object {
          "id": 4,
          "jsonb": "{\\"int\\":4,\\"string\\":\\"test\\",\\"boolean\\":false}",
        },
      ],
    },
    "jsonb_contains_boolean": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "jsonb": "{\\"int\\":1,\\"string\\":\\"TEST\\",\\"boolean\\":true}",
        },
        Object {
          "id": 2,
          "jsonb": "{\\"int\\":2,\\"string\\":\\"Test\\",\\"boolean\\":true}",
        },
      ],
    },
    "jsonb_contains_int": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "jsonb": "{\\"int\\":2,\\"string\\":\\"Test\\",\\"boolean\\":true}",
        },
      ],
    },
    "jsonb_contains_string": Object {
      "nodes": Array [
        Object {
          "id": 4,
          "jsonb": "{\\"int\\":4,\\"string\\":\\"test\\",\\"boolean\\":false}",
        },
      ],
    },
    "not_int_equalTo_2_or_string_endsWith_t": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
      ],
    },
    "not_string_equalTo_test": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
      ],
    },
    "string_containsInsensitive_t": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_contains_t": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_distinctFrom_test": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 5,
          "string": null,
        },
      ],
    },
    "string_endsWithInsensitive_t": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_endsWith_t": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_equalTo_test": Object {
      "nodes": Array [
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_isNull_false": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_isNull_true": Object {
      "nodes": Array [
        Object {
          "id": 5,
          "string": null,
        },
      ],
    },
    "string_likeInsensitive_es": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_like_es": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_notContainsInsensitive_t": Object {
      "nodes": Array [],
    },
    "string_notContains_t": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
      ],
    },
    "string_notDistinctFrom_test": Object {
      "nodes": Array [
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_notEndsWithInsensitive_t": Object {
      "nodes": Array [],
    },
    "string_notEndsWith_t": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
      ],
    },
    "string_notEqualTo_test": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
      ],
    },
    "string_notLikeInsensitive_es": Object {
      "nodes": Array [],
    },
    "string_notLike_es": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
      ],
    },
    "string_notSimilarTo_te_or_st": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
      ],
    },
    "string_notStartsWithInsensitive_t": Object {
      "nodes": Array [],
    },
    "string_notStartsWith_t": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
      ],
    },
    "string_similarTo_te_or_st": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_startsWithInsensitive_t": Object {
      "nodes": Array [
        Object {
          "id": 1,
          "string": "TEST",
        },
        Object {
          "id": 2,
          "string": "Test",
        },
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
    "string_startsWith_t": Object {
      "nodes": Array [
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
  },
}
`;

exports[`connections-filter.null-and-empty-allowed.graphql 1`] = `
Object {
  "data": Object {
    "a": Object {
      "totalCount": 5,
    },
    "b": Object {
      "totalCount": 5,
    },
    "c": Object {
      "totalCount": 5,
    },
    "d": Object {
      "totalCount": 5,
    },
    "e": Object {
      "totalCount": 5,
    },
    "f": Object {
      "totalCount": 5,
    },
    "g": Object {
      "totalCount": 5,
    },
    "h": Object {
      "totalCount": 5,
    },
    "i": Object {
      "totalCount": 5,
    },
    "j": Object {
      "totalCount": 5,
    },
    "k": Object {
      "totalCount": 5,
    },
    "l": Object {
      "totalCount": 5,
    },
    "m": Object {
      "totalCount": 5,
    },
    "n": Object {
      "totalCount": 5,
    },
    "o": Object {
      "totalCount": 5,
    },
    "p": Object {
      "totalCount": 5,
    },
    "q": Object {
      "totalCount": 5,
    },
    "r": Object {
      "totalCount": 5,
    },
    "s": Object {
      "totalCount": 5,
    },
    "t": Object {
      "totalCount": 5,
    },
  },
}
`;

exports[`connections-filter.null-and-empty-forbidden.graphql 1`] = `
Object {
  "data": Object {
    "a": null,
    "b": null,
    "c": null,
    "d": null,
    "e": null,
    "f": null,
    "g": null,
    "h": null,
    "i": null,
    "j": null,
    "k": null,
    "l": null,
    "m": null,
    "n": null,
    "o": null,
    "p": null,
    "q": null,
    "r": null,
    "s": null,
    "t": null,
  },
  "errors": Array [
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
    [GraphQLError: Null literals are forbidden in filter argument input.],
    [GraphQLError: Empty objects are forbidden in filter argument input.],
  ],
}
`;

exports[`connections-filter.relations.graphql 1`] = `
Object {
  "data": Object {
    "backwardCompound_name_equalTo_backwardCompound12": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
      ],
    },
    "backward_name_equalTo_backward2": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
      ],
    },
    "forwardCompound_name_equalTo_forwardCompound12": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
      ],
    },
    "forward_name_equalTo_forward2": Object {
      "nodes": Array [
        Object {
          "id": 2,
          "string": "Test",
        },
      ],
    },
    "parent_name_equalTo_parent2": Object {
      "nodes": Array [
        Object {
          "id": 3,
          "string": "tEST",
        },
        Object {
          "id": 4,
          "string": "test",
        },
      ],
    },
  },
}
`;

exports[`connections-filter.simple-collections.graphql 1`] = `
Object {
  "data": Object {
    "string_isNull_true": Array [
      Object {
        "id": 5,
        "string": null,
      },
    ],
  },
}
`;
